plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -142,4 +150,9 @@ public ScriptEngine getScriptEngine(Settings settings, Collection<ScriptContext<
     return new ExpressionScriptEngine(new DefaultExpressionSerializer());
   }
 
+  @Override
+  public void reload(Settings settings) {
```

Comment:
Any setting change will trigger reload, right?

Reply:
This has to be manually triggered using POST http://localhost:9200/_nodes/reload_secure_settings . When someone changes catalog settings they can trigger this api to rebuild connectors.